src/main/resources/io/viash/platforms/nextflow/WorkflowHelper.nf

```diff
@@ -603,10 +606,467 @@ def paramsToList(params, config) {
 }
 
 def paramsToChannel(params, config) {
+  System.err.println("Warning: paramsToChannel has deprecated in Viash 0.7.0. " +
+                     "Please use a combination of channelFromParams and preprocessInputs.")
   Channel.fromList(paramsToList(params, config))
 }
 
 def viashChannel(params, config) {
+  System.err.println("Warning: viashChannel has deprecated in Viash 0.7.0. " +
+                     "Please use a combination of channelFromParams and preprocessInputs.")
   paramsToChannel(params, config)
     | map{tup -> [tup.id, tup]}
 }
+
+/**
+ * From a viash config, gather the config entries 
+ * for arguments that have property 'multiple: true'
+ */
+
+private List<Map> _getMultiParameterSettings(Map config) {
+  def configArguments = config.functionality.allArguments
+  def plainNameArguments = configArguments.findAll{it.containsKey("plainName")}
+  return plainNameArguments.findAll({it.multiple})
+}
+
+/**
+ * Split parameters for arguments that accept multiple values using their seperator
+ *
+ * @param paramList A Map containing parameters to split.
+ * @param multiArgumentsSettings A List of the viash configuration argument entries 
+ *                               that have the property 'multiple: true'
+ *
+ * @return A Map of parameters where the parameter values have been split into a list using
+ *         their seperator.
+ */
+Map<String, Object> _splitParams(Map<String, Object> parValues, List<Map> multiArgumentsSettings){
+  def parsedParamValues = parValues.collectEntries { parName, parValue ->
+    parameterSettings = multiArgumentsSettings.find({it.plainName == parName})
+    if (parameterSettings) { // Check if parameter can accept multiple values
+      if (parValue instanceof List) {
+          parValue = parValue.collect{it instanceof String ? it.split(parameterSettings.multiple_sep) : it }
+      } else if (parValue instanceof String) {
+          parValue = parValue.split(parameterSettings.multiple_sep)
+      } else if (parValue == null) {
+          parValue = []
+      } else {
+          parValue = [ parValue ]
+      }
+      parValue = parValue.flatten()
+    }
+    // For all parameters check if multiple values are only passed for
+    // arguments that allow it
+    if (parValue instanceof Collection && !parameterSettings ) {
+      assert parValue.size() == 1 : 
+      "Error: argument ${parName} has too many values.\n" +
+      "  Expected amount: 1. Found: ${parValue.size()}"
+      parValue = parValue[0]
+    }
+    [parName, parValue]
+  }
+  return parsedParamValues
+}
```

Comment:
This function will throw assertion errors on parameters in `parValues` that are not in a confi's allArguments if they are length > 1 or will turn flatten the list if length == 1. I think this is an unintended consequence.

Why not simply pass the whole `config.functionality.allArguments` into the function? As an added bonus, it will make the code readable as well:

```groovy
Map<String, Object> _splitParams(Map<String, Object> parValues, List<Map> allArguments){
  def parsedParamValues = parValues.collectEntries { parName, parValue ->
    arg = allArguments.find{it.plainName == parName}

    if (!arg) {
      // if argument is not found, do not alter
      return [parName, parValue]
    }

    // Check if parameter can accept multiple values.
    // If so, make sure parValue is a list
    if (arg.multiple) {
      if (parValue instanceof Collection) {
          parValue = parValue.collect{it instanceof String ? it.split(arg.multiple_sep) : it }
      } else if (parValue instanceof String) {
          parValue = parValue.split(arg.multiple_sep)
      } else if (parValue == null) {
          parValue = [] // should we leave parValue unchanged if it is null?
      } else {
          parValue = [ parValue ]
      }
      parValue = parValue.flatten()
    }

    // For all parameters check if multiple values are only passed for
    // arguments that allow it. Quietly simplify lists of length 1.
    if (!arg.multiple && parValue instanceof Collection) {
      assert parValue.size() == 1 : 
        "Error: argument ${parName} has too many values.\n" +
        "  Expected amount: 1. Found: ${parValue.size()}"
      parValue = parValue[0]
    }

    [parName, parValue]
  }

  return parsedParamValues
}
```

Reply:
Adjusted